lib/src/benchmark_base.dart

```diff
@@ -62,3 +62,64 @@ class BenchmarkBase {
     emitter.emit(name, measure());
   }
 }
+
+class AsyncBenchmarkBase {
+  final String name;
+  final ScoreEmitter emitter;
+
+  // Empty constructor.
+  const AsyncBenchmarkBase(this.name, {this.emitter = const PrintEmitter()});
+
+  // The benchmark code.
+  // This function is not used, if both [warmup] and [exercise] are overwritten.
+  run() async {}
+
+  // Runs a short version of the benchmark. By default invokes [run] once.
+  void warmup() async {
+    await run();
+  }
+
+  // Exercices the benchmark. By default invokes [run] 10 times.
+  void exercise() async {
+    for (int i = 0; i < 10; i++) {
+      await run();
+    }
+  }
```

Comment:
Okay story time. The class in `package:benchmark_harness` is bug compatible with Golem and I need to keep them that way until I've migrated the benchmarks. Golem has an existing AsyncBenchmarkBase, and I need to keep this version compatible with it.

BenchmarkBase (including the one here) has a bug where it does ten iterations per exercise, but it doesn't multiply the final result by 10, meaning that it reports the time to run 10 iterations instead of 1. The documentation for this package mentions that.

The existing AsyncBenchmarkBase doesn't have that problem. It does one run per exercise. I need this version to be compatible with that. That's still not great because doing 10 iterations per exercise means less overhead checking the system time, but at least the results aren't misleading.

My intention is to take over this package and ensure there's a pair of sync/async base benchmark classes that report the time per iteration, and perhaps does 10 iterations per exercise.

So where does that leave you? I do need a class named AsyncBenchmarkBase that does one run per exercise and reports the time per iteration. This is results compatible with how we want things to be in the future. You can use that base class in your Isolates benchmark. When I make changes to this package in the future, I can update your benchmark while being results compatible.

Reply:
Done